johnsonandjohnson/Bodiless-JS
# feat(psh): Provide default psh cache expiry and allow local override

## Changes
-  `.platform.app.yaml` files are now merged with the local versions taking precedence.
- Only keys from `/bodiless-psh/resources/.platform/platform.whitelist.yaml` will be merged with local `.platform.app.yaml` taking precedence.
- Other keys will default to `*.platform.app.yaml` in `@bodiless/psh`

## Test Instructions
#### Test if `expires` prop works:
- Deploy a site to p.sh using updated `@bodiless/psh` package.
- Check the `network` tab in the chrome console.
- You should see the `Cache-Control` and `Expires` headers enabled.

#### Test if `.platform.app.yaml` files merged right:
- Run `npm run sites:clone-local starter`
- Run `npm run sites:update starter`
- Go into `cd sites/starter`
- Add `"bodiless-psh-init": "bodiless-psh-init"` to the `scripts` section
- Run `npm run bodiless-psh-init`
- Check if the right files were generated. ( should copy the defaults if none exists )
- Try changing `.platform.app.yaml` file based on the `platform.whitelist.yaml` and running `npm run sites:update starter` and `npm run bodiless-psh-init`

## Related Issues
Closes: https://github.com/johnsonandjohnson/Bodiless-JS/issues/172

BREAKING CHANGE: 
- Instead of just copying `*.platform.app.yaml` files it will merge existing local versions with a default version from `@bodiless/psh` with local versions taking precedence. Only keys from `/bodiless-psh/resources/.platform/platform.whitelist.yaml` will be merged
